testdata/p4tc_samples_outputs/add_entry_1_example_control_blocks.c

```diff
@@ -114,8 +114,8 @@ if (/* hdr->ipv4.isValid() */
                                 break;
                             case MAINCONTROLIMPL_IPV4_TBL_1_ACT_MAINCONTROLIMPL_SEND_NH: 
                                 {
-                                    hdr->ethernet.srcAddr = bpf_cpu_to_be64(value->u.MainControlImpl_send_nh.smac);
-                                                                        hdr->ethernet.dstAddr = ntohll(value->u.MainControlImpl_send_nh.dmac << 16);
+                                    storePrimitive64(&hdr->ethernet.srcAddr, 48, (bpf_cpu_to_be64(value->u.MainControlImpl_send_nh.smac)));
```

Comment:
I think I accidentally found a bug
Was testing changing `EthernetAddress'`  typecast[1] to a `bit<39>` and saw that the generated code was outputing this line as:

```
storePrimitive64(&hdr->ethernet.srcAddr, 40, (bpf_cpu_to_be64((u64)(u64)value->u.MainControlImpl_send_nh.smac)));
```

From what I understood, the second argument should be `39`, not `40`

[1] https://github.com/p4lang/p4c/blob/main/testdata/p4tc_samples/add_entry_1_example.p4#L4

Reply:
I am unable to reproduce this issue, If I change srcaddr to bit<39>, I get error about invalid ethernet packet size.